packages/api/src/auth/auth.service.ts

```diff
@@ -0,0 +1,32 @@
+import { ConflictException, Injectable } from '@nestjs/common';
+import { EventBus } from '@nestjs/cqrs';
+import { Prisma } from '@prisma/client';
+
+import { registerError } from '@coderscamp/shared/models/auth/register';
+import { pick } from '@coderscamp/shared/utils/object';
+
+import { AuthRepository } from './auth.repository';
+import { UserRegistrationCompletedEvent } from './events/user-registration-completed.event';
+import { hashPassword } from './local/local.utils';
+
+@Injectable()
+export class AuthService {
+  constructor(private readonly eventBus: EventBus, private readonly authRepository: AuthRepository) {}
+
+  async register(data: Prisma.UserRegistrationCreateInput & Prisma.AuthUserCreateInput) {
```

Comment:
This is dangerous. Database logic spreads out all app layers.

Reply:
I'd pass to register some object like RegisterUserCommand and encapsulate all things related to db inside the service.